qiskit/circuit/quantumcircuit.py

```diff
@@ -240,7 +246,8 @@ def data(self, data_input):
             self.append(inst, qargs, cargs)
 
     def __str__(self):
-        return str(self.draw(output='text'))
+        from qiskit.visualization import circuit_drawer
+        return str(circuit_drawer(self, output='text'))
```

Comment:
This change is pointing to why this is not a good idea. We don't want to change the return type based on the environment you run in.